### ISSUE TITLE
group_similar_strings with series length 1 produces ValueError

### ISSUE THREAD

**Issue by gw00207** [Bergvca/string_grouper#52]
using the group_similar_strings method in version 0.4.0, if a series of length 1 is sent, a ValueError with no explanation is raised. To reproduce:

`group_similar_strings(strings_to_group=pd.Series(['hello']))`

I believe in previous versions this wasn't the case. I would have expected a DataFrame such as:

| group_rep_index      | group_rep |
| ----------- | ----------- |
| 0      | hello       |

full stacktrace:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-18-f6cdcfd0b8c6> in <module>
----> 1 group_similar_strings(strings_to_group=pd.Series(['hello']))

~/.cache/pypoetry/virtualenvs/env/lib/python3.8/site-packages/string_grouper/string_grouper.py in group_similar_strings(strings_to_group, string_ids, **kwargs)
     78     """
     79     string_grouper = StringGrouper(strings_to_group, master_id=string_ids, **kwargs).fit()
---> 80     return string_grouper.get_groups()
     81 
     82 

~/.cache/pypoetry/virtualenvs/env/lib/python3.8/site-packages/string_grouper/string_grouper.py in wrapper(*args, **kwargs)
    181     def wrapper(*args, **kwargs):
    182         if args[0].is_build:
--> 183             return f(*args, **kwargs)
    184         else:
    185             raise StringGrouperNotFitException(f'{f.__name__} was called before the "fit" function was called.'

~/.cache/pypoetry/virtualenvs/env/lib/python3.8/site-packages/string_grouper/string_grouper.py in get_groups(self, ignore_index, replace_na)
    366         if ignore_index is None: ignore_index = self._config.ignore_index
    367         if self._duplicates is None:
--> 368             return self._deduplicate(ignore_index=ignore_index)
    369         else:
    370             if replace_na is None: replace_na = self._config.replace_na

~/.cache/pypoetry/virtualenvs/env/lib/python3.8/site-packages/string_grouper/string_grouper.py in _deduplicate(self, ignore_index)
    606         # pandas groupby transform function and enlargement enable both respectively in one step:
    607         group_of_master_index['group_rep'] = \
--> 608             group_of_master_index.groupby('raw_group_id', sort=False)['weight'].transform(method)
    609 
    610         # Prepare the output:

~/.cache/pypoetry/virtualenvs/env/lib/python3.8/site-packages/pandas/core/groupby/generic.py in transform(self, func, *args, **kwargs)
    475         # result to the whole group. Compute func result
    476         # and deal with possible broadcasting below.
--> 477         result = getattr(self, func)(*args, **kwargs)
    478         return self._transform_fast(result, func)
    479 

~/.cache/pypoetry/virtualenvs/env/lib/python3.8/site-packages/pandas/core/groupby/groupby.py in wrapper(*args, **kwargs)
    654             if self.obj.ndim == 1:
    655                 # this can be called recursively, so need to raise ValueError
--> 656                 raise ValueError
    657 
    658             # GH#3688 try to operate item-by-item

ValueError: 
```

**Comment by ParticularMiner** [comment]
Many thanks George for pointing out this bug!  
Please find the fix in PR #53 .

**Comment by gw00207** [comment]
thanks for resolving so quickly 🚀
